horace_core/sqw/@sqw/sqw_eval.m

```diff
@@ -1,102 +1,348 @@
+%                   weight = sqwfunc (qh, qk, ql, en, p)
 %                where
 %                   qh,qk,ql,en Arrays containing the coordinates of a set of points
 %                   p           Vector of parameters needed by dispersion function
-%                              e.g. [A,js,gam] as intensity, exchange, lifetime
+%                              e.g. [A, js, gam] as intensity, exchange, lifetime
 %                   weight      Array containing calculated spectral weight
 %
 %               More general form is:
-%                   weight = sqwfunc (qh,qk,ql,en,p,c1,c2,..)
+%                   weight = sqwfunc (qh, qk, ql, en, p, c1, c2, ..)
 %                 where
 %                   p           Typically a vector of parameters that we might want
 %                              to fit in a least-squares algorithm
-%                   c1,c2,...   Other constant parameters e.g. file name for look-up
+%                   c1, c2, ...   Other constant parameters e.g. file name for look-up
 %                              table
 %
 %   pars       Arguments needed by the function. Most commonly, a vector of parameter
-%              values e.g. [A,js,gam] as intensity, exchange, lifetime. If a more general
+%              values e.g. [A, js, gam] as intensity, exchange, lifetime. If a more general
 %              set of parameters is required by the function, then
 %              package these into a cell array and pass that as pars. In the example
 %              above then pars = {p, c1, c2, ...}
 %
-% Optional string parameters:
-%   'all'      Requests that the calculated sqw be returned over
+% Keyword Arguments:
+% ------------------
+%   outfile    If present, the outputs will be written to the file of the given
+%              name/path.
+%              If numel(win) > 1, outfile must be omitted or a cell array of
+%              file paths with equal number of elements as win.
+%
+%   all        Requests that the calculated sqw be returned over
 %              the whole of the domain of the input dataset. If not given, then
 %              the function will be returned only at those points of the dataset
 %              that contain data.
-%               Applies only to input with no pixel information - it is ignored if
+%              Applies only to input with no pixel information - it is ignored if
 %              full sqw object.
 %
-%   'average' Requests that the calculated sqw be computed for the
-%              average values of h,k,l of the pixels in a bin, not for each
+%   average    Requests that the calculated sqw be computed for the
+%              average values of h, k, l of the pixels in a bin, not for each
 %              pixel individually. Reduces cost of expensive calculations.
-%              Applies only to the case of sqw object with pixel information 
+%              Applies only to the case of sqw object with pixel information
 %             - it is ignored if dnd type object.
```

Comment:
I've added that we expect logical arguments. 

In [ADR 22](https://github.com/pace-neutrons/Horace/blob/master/documentation/adr/0022-use-keyword-args.md) we decided that we would use `-flags` and boolean `'name', value` pairs interchangeably. The usage example at the top of the help should demonstrate this.

Can I suggest we open an issue to update the `dnd` function helps separately? I know that Nick has re-factored the `dnd` implementation into the base class, so it would make more sense to do it there.